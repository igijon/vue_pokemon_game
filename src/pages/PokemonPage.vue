<template>
  <h1>¿Quién es este Pokémon?</h1>

  <PokemonPicture :pokemonId="151" :showPokemon="true"/>
  <PokemonOptions/>
</template>

<script>
import PokemonOptions from '../components/PokemonOptions.vue'
import PokemonPicture from '../components/PokemonPicture.vue'

import getPokemonOptions from '@/helpers/getPokemonOptions'

getPokemonOptions()

export default {
    name: 'PokemonPage',
    components: {
        PokemonOptions,
        PokemonPicture
    }
}
</script>

<style>

</style>