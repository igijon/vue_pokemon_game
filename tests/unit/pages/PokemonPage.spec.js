import { shallowMount, mount } from '@vue/test-utils' 
import PokemonPage from '@/pages/PokemonPage'

describe('PokemonPage component', () => {

    let wrapper
    let mixPokemonArraySpy

    beforeEach(() => {
        jest.clearAllMocks()
        mixPokemonArraySpy = jest.spyOn( PokemonPage.methods, 'mixPokemonArray' )
        wrapper = shallowMount( PokemonPage )

    })

    /*
    No merece la pena el match con el snapshot porque si no hay pokemon sÃ³lo renderiza el h2 de espere por favor
    */
    test('debe hacer match con el snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot()
    })

    test('debe llamar mixPokemonArray al montar', () => {

        expect( mixPokemonArraySpy ).toHaveBeenCalled()

    })
})