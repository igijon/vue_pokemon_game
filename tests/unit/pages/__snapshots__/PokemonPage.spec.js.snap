// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPage component debe hacer match con el snapshot 1`] = `
<h1>Espere por favor...</h1>
<!--Si no quiero un div adicional por motivos de diseÃ±o para que no aparezca en el
  html puedo sustituirlo por template-->
<!--v-if-->
`;
